apis/Google.Devtools.AspNet/Google.Devtools.AspNet.Tests/CloudErrorReporting/CloudErrorReportingExceptionLoggerTest.cs

```diff
@@ -0,0 +1,166 @@
+            requestMessage.Method = DeleteMethod;
+            requestMessage.RequestUri = GoogleUri;
+            requestMessage.Headers.UserAgent.Add(UserAgentValue);
+
+            HttpResponseMessage responseMessage = new HttpResponseMessage();
+            responseMessage.StatusCode = ConflictStatusCode;
+
+            Exception exception;
+            try
+            {
+                throw SimpleException;
+            }
+            catch (Exception e)
+            {
+                exception = e;
+            }
+
+            ExceptionContext exceptionContext = new ExceptionContext(
+                exception, ExceptionCatchBlocks.HttpServer, requestMessage, responseMessage);
+            return new ExceptionLoggerContext(exceptionContext);
+        }
+
+        /// <summary>
+        /// Matcher to check if a <see cref="ReportedErrorEvent"/> matches one 
+        /// generated by <see cref="CreateComplexContext"/>
+        /// </summary>
+        private ReportedErrorEvent IsComplexContext()
+        {
+            return Match.Create<ReportedErrorEvent>(e =>
+                e.Message.Contains(SimpleException.Message) &&
+                e.Context.HttpRequest.Method.Equals(DeleteMethod.ToString()) &&
+                e.Context.HttpRequest.Url.Equals(GoogleUri.ToString()) &&
+                e.Context.HttpRequest.UserAgent.Equals(UserAgentValue.ToString()) &&
+                e.Context.HttpRequest.ResponseStatusCode == (int)ConflictStatusCode &&
+                e.Context.ReportLocation.LineNumber > 0 &&
+                !string.IsNullOrEmpty(e.Context.ReportLocation.FilePath) &&
+                e.Context.ReportLocation.FunctionName.Equals("CreateComplexContext") &&
+                e.ServiceContext.Service.Equals(ServiceName) &&
+                e.ServiceContext.Version.Equals(Version)
+            );
+        }
+
+        /// <summary>
+        /// Matcher to check if a <see cref="ReportedErrorEvent"/> matches
+        /// the <see cref="SimpleContext"/>
+        /// </summary>
+        private ReportedErrorEvent IsSimpleContext()
+        {
+            return Match.Create<ReportedErrorEvent>(e =>
+                e.Message.Contains(SimpleException.Message) &&
+                string.IsNullOrEmpty(e.Context.HttpRequest.Method) &&
+                string.IsNullOrEmpty(e.Context.HttpRequest.Url) &&
+                string.IsNullOrEmpty(e.Context.HttpRequest.UserAgent) &&
+                e.Context.HttpRequest.ResponseStatusCode == 0 &&
+                e.Context.ReportLocation.LineNumber == 0 &&
+                string.IsNullOrEmpty(e.Context.ReportLocation.FilePath) &&
+                string.IsNullOrEmpty(e.Context.ReportLocation.FunctionName) &&
+                e.ServiceContext.Service.Equals(ServiceName) &&
+                e.ServiceContext.Version.Equals(Version)
+            );
+        }
+
+        private Mock<ReportErrorsServiceClient> GetClientMock()
```

Comment:
I think I'd personally just inline this in each usage - it makes each test simpler to read by depending on fewer things, and we're not actually using any logic that would need duplication.